aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -533,7 +533,7 @@ Tensor frobenius_norm(const Tensor& self, IntArrayRef dim, bool keepdim) {
     return at::norm(self, 2, dim, keepdim, self.scalar_type());
   }
   if (self.is_complex()){
-    return at::sqrt(at::sum((self.conj() * self).real(), dim, keepdim));
+    return at::sqrt(at::sum(at::real(self.conj() * self), dim, keepdim));
```

Comment:
this changes the return type of norm, right?  Is that what we want?

Reply:
perhaps a less risky way of doing this would be to create an underscore function like:

`at::_real_complex` that does exactly what `real` does today and switch all usages over to that new form.

Reply:
The current change puts the code in a misleading state. This change will break norm when called on complex tensors -- it will throw a runtime error when it hits real. This is good and bad:

1) good - we want to change how this behavior works
2) bad - it's ugly

We could, instead, add nicer error messages to functions like norm that they're not implemented for complex inputs. That would add a little work for someone later to remove.

We could add at::_real_complex but that seems like trying to kill ants with a hammer?